apps/settings/js/modules/bluetooth/bluetooth_context.js

```diff
@@ -824,6 +875,177 @@ define(function(require) {
         Debug('unpair(): Reject with this reason: ' + reason);
         return Promise.reject(reason);
       });
+    },
+
+    /**
+     * Init the connecting device which is browsed in paired devices list. 
+     * Get connection info from ConnectionManager.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     */
+    _initConnectingDevices: function btc__initConnectingDevices() {
+      // Init the paired device connection status for connecting device.
+      if (!ConnectionManager.connectingAddress) {
+        return;
+      }
+
+      var existedDevice = 
+        this._findDeviceByAddress({
+          paired: true, 
+          address: ConnectionManager.connectingAddress
+        });
+      if (existedDevice) {
+        // The connecting device is existed. Init connection status for it.
+        var options = {
+          connectionStatus: 'connecting'
+        };
+        Debug('_initConnectingDevices(): options = ' + JSON.stringify(options));
+        existedDevice.updateConnectionInfo(options);
+      }
+    },
+
+    /**
+     * Init the connected device which is browsed in paired devices list.
+     * Get connection info from ConnectionManager.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     */
+    _initConnectedDevices: function btc__initConnectedDevices() {
+      // Init the paired device connection status for connected device.
+      var connectedDevices = ConnectionManager.getConnectedDevices();
+      for (var address in connectedDevices) {
+        var existedDevice = 
+          this._findDeviceByAddress({paired: true, address: address});
+        
+        if (existedDevice) {
+          // The connected device is existed. 
+          // Init connection status/profiles for it.
+          var options = {
+            connectionStatus: 'connected',
+            profiles: connectedDevices[address].connectedProfiles
+          };
+          Debug('_initConnectedDevices(): address = ' + address + 
+                ', options = ' + JSON.stringify(options));
+          existedDevice.updateConnectionInfo(options);
+        }
+      }
+    },
+
+    /**
+     * Device 'connectionStatus' property is changed from 
+     * ConnectionManager operation.
+     * Update device 'connectionStatus' property via reference address in event.
+     *
+     * @access private
+     * @memberOf BluetoothContext
+     * @param {Object} event
+     * @param {String} event.address - address of device
+     */
+    _updateDeviceConnectingStatus: 
```

Comment:
Note that there are only slight difference among `_updateDeviceConnectingStatus`, `_updateDeviceConnectedStatus`, `_updateDeviceDisconnectedStatus`, and `_updateDeviceConnectionProfile`. To be more specific, the only difference is the options object to be passed to the update function.

We can combine these four functions into one. The key is that the event object should have a property called 'type' allowing you to differentiate different kinds of events.